core/arch/arm/sm/psci.c

```diff
@@ -149,9 +149,7 @@ void tee_psci_handler(struct thread_smc_args *args)
 			;
 		break;
 	case PSCI_SYSTEM_RESET:
-		psci_system_off();
-		while (1)
-			;
+		psci_system_reset();
```

Comment:
The function isn't supposed to return at all, but if it does nonetheless I assume that returning  some error code like `PSCI_RET_INTERNAL_FAILURE` instead of `0x84000009` would be preferable.